src/AltinnCore/Common/Helpers/InstantiationHelper.cs

```diff
@@ -0,0 +1,132 @@
+using System;
+using System.Collections.Generic;
+using Altinn.Platform.Storage.Models;
+using AltinnCore.ServiceLibrary.Enums;
+using AltinnCore.ServiceLibrary.Models;
+
+namespace Common.Helpers
+{
+    /// <summary>
+    /// Various helper methods for instantiation
+    /// </summary>
+    public static class InstantiationHelper
+    {
+        private const string BANKRUPTCY_CODE = "KBO";
+        private const string SUB_UNIT_CODE = "BEDR";
+        private const string SUB_UNIT_CODE_AAFY = "AAFY";
+
+        /// <summary>
+        /// Filters a list of parties based on an applications allowed party types.
+        /// </summary>
+        /// <param name="parties">The list of parties to be filtered</param>
+        /// <param name="partyTypesAllowed">The allowed party types</param>
+        /// <returns>A list with the filtered parties</returns>
+        public static List<Party> FilterPartiesByAllowedPartyTypes(List<Party> parties, PartyTypesAllowed partyTypesAllowed)
+        {
+            List<Party> allowed = new List<Party>();
+            if (parties == null || partyTypesAllowed == null)
+            {
+                return allowed;
+            }
+
+            parties.ForEach(party =>
+            {
+                if (IsPartyAllowedToInstantiate(party, partyTypesAllowed))
+                {
+                    allowed.Add(party);
+                }
+            });
+            return allowed;
+        }
+
+        /// <summary>
+        /// Checks if a party is allowed to initiate an application based on the applications AllowedPartyTypes
+        /// </summary>
+        /// <param name="party">The party to check</param>
+        /// <param name="partyTypesAllowed">The allowed party types</param>
+        /// <returns>True or false</returns>
+        public static bool IsPartyAllowedToInstantiate(Party party, PartyTypesAllowed partyTypesAllowed)
+        {
+            if (party == null)
+            {
+                return false;
+            }
+
+            if (partyTypesAllowed == null || (!partyTypesAllowed.BankruptcyEstate && !partyTypesAllowed.Organization && !partyTypesAllowed.Person && !partyTypesAllowed.SubUnit))
+            {
+                // if party types not set, all parties are allowed to initiate
+                return true;
+            }
+
+            PartyType partyType = party.PartyTypeName;
+            bool isAllowed = false;
+            switch (partyType)
+            {
+                case PartyType.Person:
+                    if (partyTypesAllowed.Person == true)
+                    {
+                        isAllowed = true;
+                    }
+
+                    break;
+                case PartyType.Organization:
+                    if (partyTypesAllowed.Organization == true)
+                    {
+                        isAllowed = true;
+                    }
+                    else if (party.UnitType == null)
```

Comment:
I'm guessing this is to make sure there is a unit type defined before checking the remaining statements.. Consider adding the check to the respective statements to make it easier to understand or encase them in a seperate if-statement to make it easier to read the code.

Reply:
Good point. Fixed.